quantmsio/core/quantms/psm.py

```
@@ -53,15 +55,18 @@ def iter_psm_table(
             yield df
 
     def _extract_pep_columns(self) -> None:
-        if os.stat(self.mztab_path).st_size == 0:
+        if self.mztab_path.stat().st_size == 0:
             raise ValueError("File is empty")
-        f = open(self.mztab_path)
+        # Use seekable file path to handle gzip files
+        seekable_path = self._get_seekable_file_path()
+        f = open(seekable_path, "r", encoding="utf-8")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)